lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -0,0 +1,581 @@
+            self._axis = self.yaxis
+            self._locstrings = ['right', 'left']
+            self._otherstrings = ['top', 'bottom']
+        # this gets positioned w/o constrained_layout so exclude:
+        self._layoutbox = None
+        self._poslayoutbox = None
+
+        self.set_location(location)
+        self.set_conversion(conversion)
+
+        # styling:
+        if self._orientation == 'x':
+            otheraxis = self.yaxis
+        else:
+            otheraxis = self.xaxis
+
+        otheraxis.set_major_locator(mticker.NullLocator())
+        otheraxis.set_ticks_position('none')
+
+        for st in self._otherstrings:
+            self.spines[st].set_visible(False)
+        for st in self._locstrings:
+            self.spines[st].set_visible(True)
+
+        if self._pos < 0.5:
+            # flip the location strings...
+            self._locstrings = self._locstrings[::-1]
+        self.set_axis_orientation(self._locstrings[0])
+
+    def set_axis_orientation(self, orient):
+        """
+        Set if axes spine and labels are drawn at top or bottom of the
+        axes.
+
+        Parameters
+        ----------
+        orient :: string
+            either 'top' or 'bottom'
+
+        """
+        if orient in self._locstrings:
+            if orient == self._locstrings[1]:
+                # need to change the orientation.
+                self._locstrings = self._locstrings[::-1]
+            elif orient != self._locstrings[0]:
+                warnings.warn('"{}" is not a valid axis orientation, '
+                            'not changing the orientation;'
+                            'choose "{}" or "{}""'.format(orient,
+                            self._locstrings[0], self._locstrings[1]))
+            self.spines[self._locstrings[0]].set_visible(True)
+            self.spines[self._locstrings[1]].set_visible(False)
+            self._axis.set_ticks_position(orient)
+            self._axis.set_label_position(orient)
+
+    def set_location(self, location):
+        """
+        Set the vertical location of the axes in parent-normalized
+        co-ordinates.
+
+        Parameters
+        ----------
+        location : string or scalar
+            The position to put the secondary axis.  Strings can be 'top' or
+            'bottom', scalar can be a float indicating the relative position
+            on the parent axes to put the new axes, 0 being the bottom, and
+            1.0 being the top.
+        """
+
+        # This puts the rectangle into figure-relative coordinates.
+        if isinstance(location, str):
+            if location in ['top', 'right']:
+                self._pos = 1.
+            elif location in ['bottom', 'left']:
+                self._pos = 0.
+            else:
+                warnings.warn("location must be '{}', '{}', or a "
+                                  "float, not '{}'.".format(location,
+                                  self._locstrings[0], self._locstrings[1]))
```

Comment:
It looks like you have the `format` arguments out of order; `location` should be last, not first, shouldn't it?